odoo/models.py

```diff
@@ -2572,8 +2587,122 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
 
         self._read_group_format_result(rows_dict, lazy_groupby)
 
+        self._read_group_format_properties_result(rows_dict, lazy_groupby, domain)
+
         return rows_dict
 
+    def _read_group_format_properties_result(self, data, fields, domain):
+        """Modify the final read group properties result.
+
+        Replace the relational properties ids by a lazy name get, replace the "raw" tags
+        and selection values by a list containing their labels. Adapt the domains for
+        the Falsy group (we can't just keep (selection, =, False) e.g. because some values
+        in database might correspond to  option that have been remove on the parent).
+        """
+        fields = [fname.split(':')[0] for fname in fields]
+        properties_fields = filter(lambda fname: self._fields[fname.split(".")[0]].type == 'properties', fields)
+
+        for fullname in properties_fields:
+            if '.' not in fullname:
+                raise UserError(_('You must choose the property you want to group by.'))
+
+            definition = self.get_property_definition(fullname)
+            property_type = definition.get('type')
+
+            if property_type == 'selection':
+                options = definition.get('selection') or []
+                options = {option[0]: option[1] for option in options}
+                for d in data:
+                    if not d[fullname]:
+                        # can not do ('selection', '=', False) because we might have
+                        # option in database that does not exist anymore
+                        d['__domain'] = expression.AND([
```

Comment:
Whut ? `read_group` is not just for the list/kanban views. The rest of your code manages multiple groups, it is sad to be restricted here.

What about all the other view types (graph/pivot/gantt/...) ? (It crashes for graph by example :cry: )

Reply:
it crash for a different reason no ?

(they changed something and i didn't catch it after my many rebases...)

When I fixed the issue it work properly for many group, not sur to understand your point sorry


![screenshot_spect](https://github.com/odoo/odoo/assets/54316226/54e40454-b871-482a-992a-1b9220f27d6e)

Reply:
The `__domain` is useful when you click on a group itself to show records in this group. 
In your case when `d[fullname]` is Falsy, it will rewrite `__domain` without taking in account `__domain` generates by a other groupby clause.

Here a example, groupby a property (many2one) and partner_id (normal field):
```JSON
[
  {
    "lead_properties.2b7505616fa5903a": false,
    "partner_id": [
      148,
      "Other"
    ],
    "__count": 1,
    "__domain": [
      "&",
      "|",
      [
        "lead_properties.2b7505616fa5903a",
        "=",
        false
      ],
      [
        "lead_properties.2b7505616fa5903a",
        "not in",
        []
      ],
      [
        "type",
        "=",
        "opportunity"
      ]
    ]
  },
  {
    "lead_properties.2b7505616fa5903a": false,
    "partner_id": [
      147,
      "STD"
    ],
    "__count": 1,
    "__domain": [
      "&",
      "|",
      [
        "lead_properties.2b7505616fa5903a",
        "=",
        false
      ],
      [
        "lead_properties.2b7505616fa5903a",
        "not in",
        []
      ],
      [
        "type",
        "=",
        "opportunity"
      ]
    ]
  }
]
```
You can see here that the `__domain` is wrong, it doesn't take in account the partner_id. Which leads to very weird behavior when you click on the group.

```Python
                       d['__domain'] = expression.AND([
                            expression.OR([
                                [(fullname, '=', False)],
                                [(fullname, 'not in', tuple(options))],
                            ]),
                            d['__domain'],
                        ])
```

Can you add some tests to cover multi-groupby?